stencil-library/src/components/form/form-control-monetary.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  Component,
+  Host,
+  h,
+  Prop,
+  Event,
+  EventEmitter,
+  Element,
+  Watch,
+} from '@stencil/core';
+import IMask, { InputMask } from 'imask';
+
+@Component({
+  tag: 'form-control-monetary',
+  styleUrl: 'form-control-number.scss',
+  shadow: true,
+})
+export class MonetaryInput {
+  @Element() el: HTMLElement;
+
+  @Prop() label: string;
+  @Prop() name: any;
+  @Prop() error: string;
+  @Prop() defaultValue: string;
+  @Prop() inputHandler: (name: string, value: string) => void;
+
+  private imask: InputMask<any> | null = null;
+
+  @Event() formControlInput: EventEmitter<any>;
+  @Event() formControlBlur: EventEmitter<any>;
+
+  @Watch('defaultValue')
+  handleDefaultValueChange(newValue: string) {
+    this.updateInput(newValue);
+  }
+
+  componentDidLoad() {
+    const inputElement = this.el.shadowRoot.querySelector('input');
```

Comment:
Can we use a `ref` instead? I believe it's supposed to be more reliable: https://stenciljs.com/docs/templating-jsx#getting-a-reference-to-a-dom-element

Reply:
Oh sure, I forgot about that!